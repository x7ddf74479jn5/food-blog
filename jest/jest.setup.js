// Optional: configure or set up a testing framework before each test.
// If you delete this file, remove `setupFilesAfterEnv` from `jest.config.js`

// Used for __tests__/testing-library.js
// Learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom/extend-expect";

window.matchMedia = jest.fn().mockImplementation((query) => {
  return {
    addListener: jest.fn(),
    matches: false,
    media: query,
    onchange: null,
    removeListener: jest.fn(),
  };
});
