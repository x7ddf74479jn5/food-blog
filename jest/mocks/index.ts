export { mockNextImage } from "./image";
export type { NextRouter } from "./router";
export { mockRouter, RouterContext, withMockedRouter } from "./router";
export { setupWindow } from "./window";
