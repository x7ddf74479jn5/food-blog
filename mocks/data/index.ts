export { mockArticles } from "./articles";
export { mockCategories } from "./categories";
export { mockConfig } from "./config";
export { mockPickup } from "./pickup";
export { mockTags } from "./tags";
export { dateCommon } from "./utils";
export { mockWriters } from "./writers";
