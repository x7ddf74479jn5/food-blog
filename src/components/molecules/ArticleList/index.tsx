import NextLink from "@/components/atoms/NextLink";
import Thumbnail from "@/components/atoms/Thumbnail";
import type { TArticle } from "@/types";

type Props = {
  articles: TArticle[];
};

const ArticleList: React.VFC<Props> = ({ articles }) => {
  return (
    <div className="space-y-12">
      {articles.map((article) => (
        <div key={article.id}>
          <div className="mb-4">
            <Thumbnail src={article.image.url} title={article.title} id={article.id} />
          </div>

          <h2 className="mb-4 text-2xl font-bold">
            <NextLink href={`/articles/${article.id}`}>{article.title}</NextLink>
          </h2>

          <p className="dark:text-gray-300">{article.description}</p>
        </div>
      ))}
    </div>
  );
};

export default ArticleList;
