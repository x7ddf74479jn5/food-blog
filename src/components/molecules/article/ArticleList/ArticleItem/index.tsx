import { memo } from "react";

import NextLink from "@/components/atoms/NextLink";
import Thumbnail from "@/components/atoms/Thumbnail";
import type { TArticle } from "@/types";
import { urlTable } from "@/utils/paths/url";

type ArticleItem = {
  article: TArticle;
  isHighPriority?: boolean;
};

export const ArticleItem: React.FC<ArticleItem> = memo(({ article, isHighPriority }) => {
  return (
    <article className="mx-auto max-w-lg">
      <div className="mb-4">
        <Thumbnail
          src={article.image.url}
          title={article.title}
          id={article.id}
          blurDataURL={article.image.url}
          alt={article.title}
          loading={isHighPriority ? "eager" : "lazy"}
          priority={isHighPriority}
        />
      </div>
      <h2 className="mb-4 text-2xl font-bold">
        <NextLink href={`${urlTable.articles}/${article.id}`}>{article.title}</NextLink>
      </h2>
      <p className="line-clamp-2 dark:text-gray-300">{article.description}</p>
    </article>
  );
});

ArticleItem.displayName = "ArticleItem";
