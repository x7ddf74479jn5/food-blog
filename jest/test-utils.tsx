import type Queries from "@testing-library/dom/types/queries";
import type { RenderResult } from "@testing-library/react";
import { render } from "@testing-library/react";
import type { RequestHandler } from "msw";
import { setupServer } from "msw/node";
import { SWRConfig } from "swr";

import { SearchProvider } from "@/contexts/search/SearchContext";
import { ViewportProvider } from "@/contexts/viewport/ViewportContext";

import { withMockLegacyRouter, withMockRouter } from "./mocks";

export const Providers: React.ComponentType<{ children?: React.ReactNode }> = ({ children }) => {
  return withMockRouter(
    withMockLegacyRouter(
      <SWRConfig value={{ dedupingInterval: 0, provider: () => new Map() }}>
        <SearchProvider>
          <ViewportProvider>{children}</ViewportProvider>
        </SearchProvider>
      </SWRConfig>
    )
  );
};

const customRender = (ui: React.ReactElement, options = {}): RenderResult<typeof Queries, HTMLElement> => {
  return render(ui, { wrapper: Providers, ...options });
};

// re-export everything
export * from "./mocks";
export * from "@testing-library/react";

// override render method
export { render as originalRender, customRender as render };

export const reTestCase = {
  anyImage: expect.stringMatching(/^(data:image\/gif)|\.(png|webp|jpeg|jpg|svg)$/),
  anyWord: expect.stringMatching(/\w+/),
};

type ErrorComponentProps = {
  message?: string;
};
export const ErrorComponent: React.FC<ErrorComponentProps> = ({ message }) => {
  throw new Error(message);
};

export const setupMockServer = (...handlers: RequestHandler[]) => {
  const server = setupServer(...handlers);
  beforeAll(() => server.listen());
  afterEach(() => server.resetHandlers());
  afterAll(() => server.close());
  return server;
};
