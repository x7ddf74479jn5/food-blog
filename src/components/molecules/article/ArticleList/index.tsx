import { memo } from "react";

import type { TArticle } from "@/types";

import { ArticleItem } from "./ArticleItem";

type ArticleListProps = {
  articles: TArticle[];
};

const ArticleList: React.FC<ArticleListProps> = ({ articles }) => {
  return (
    <ul className="space-y-12">
      {articles.map((article) => (
        <li key={article.id}>
          <ArticleItem article={article} />
        </li>
      ))}
    </ul>
  );
};

export default memo(ArticleList);
