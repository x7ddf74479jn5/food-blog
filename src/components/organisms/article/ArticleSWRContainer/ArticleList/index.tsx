import { memo } from "react";

import type { TArticle } from "@/types";

import { ArticleItem } from "./ArticleItem";

type ArticleListProps = {
  articles: TArticle[];
};

export const ArticleList: React.FC<ArticleListProps> = memo(({ articles }) => {
  return (
    <ul className="space-y-12">
      {articles.map((article) => (
        <li key={article.id}>
          <ArticleItem article={article} />
        </li>
      ))}
    </ul>
  );
});

ArticleList.displayName = "ArticleList";
