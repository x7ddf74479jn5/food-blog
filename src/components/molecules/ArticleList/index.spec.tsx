import { mockArticles } from "@mocks/data";
import { render, screen } from "jest/test-utils";

import ArticleList from "../ArticleList";

describe("components/molecules/ArticleList", () => {
  const mockArticleList = Object.values(mockArticles);

  it("OK: 表示が正しい", () => {
    render(<ArticleList articles={mockArticleList} />);
    const articles = screen.getAllByRole("article");
    expect(articles.length).toBe(mockArticleList.length);
  });
});
