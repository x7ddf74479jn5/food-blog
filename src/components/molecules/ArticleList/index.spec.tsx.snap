// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/molecules/ArticleList snapshot 1`] = `
<ul
  className="space-y-12"
>
  <li>
    <article>
      <div
        className="mb-4"
      >
        <a
          aria-label="基本の一番だしの作り方"
          href="/articles/1"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "width": "initial",
              }
            }
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            >
              <img
                alt=""
                aria-hidden={true}
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjQwIiBoZWlnaHQ9IjM2MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              />
            </span>
            <img
              alt="Cover Image for 基本の一番だしの作り方"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": "cover",
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
            <noscript>
              <img
                alt="Cover Image for 基本の一番だしの作り方"
                data-nimg="intrinsic"
                decoding="async"
                loading="lazy"
                src="/_next/image?url=%2Fimages%2F5026416_s.jpg&w=1920&q=75"
                srcSet="/_next/image?url=%2Fimages%2F5026416_s.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2F5026416_s.jpg&w=1920&q=75 2x"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": "cover",
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
            </noscript>
          </span>
        </a>
      </div>
      <h2
        className="mb-4 text-2xl font-bold"
      >
        <a
          href="/articles/1"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          基本の一番だしの作り方
        </a>
      </h2>
      <p
        className="dark:text-gray-300 line-clamp-2"
      >
        だし昆布と鰹節（削り節）から取った出汁は、味噌汁、そば、うどん、煮物、鍋料理など、多くの和食に使用できます。
      </p>
    </article>
  </li>
  <li>
    <article>
      <div
        className="mb-4"
      >
        <a
          aria-label="新玉ねぎとトマトのシンプルサラダの作り方"
          href="/articles/2"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "width": "initial",
              }
            }
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            >
              <img
                alt=""
                aria-hidden={true}
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjQwIiBoZWlnaHQ9IjM2MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              />
            </span>
            <img
              alt="Cover Image for 新玉ねぎとトマトのシンプルサラダの作り方"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": "cover",
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
            <noscript>
              <img
                alt="Cover Image for 新玉ねぎとトマトのシンプルサラダの作り方"
                data-nimg="intrinsic"
                decoding="async"
                loading="lazy"
                src="/_next/image?url=%2Fimages%2F2996666_s.jpg&w=1920&q=75"
                srcSet="/_next/image?url=%2Fimages%2F2996666_s.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2F2996666_s.jpg&w=1920&q=75 2x"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": "cover",
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
            </noscript>
          </span>
        </a>
      </div>
      <h2
        className="mb-4 text-2xl font-bold"
      >
        <a
          href="/articles/2"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          新玉ねぎとトマトのシンプルサラダの作り方
        </a>
      </h2>
      <p
        className="dark:text-gray-300 line-clamp-2"
      >
        スライストマトに、たっぷりのサラダ玉ねぎと大葉を乗せ、オリーブオイル、ポン酢、醤油、にんにくで作った特製ドレッシングをかけるだけの簡単・極旨サラダです。
      </p>
    </article>
  </li>
  <li>
    <article>
      <div
        className="mb-4"
      >
        <a
          aria-label="栄養たっぷり小松菜と油揚げのおひたしの作り方"
          href="/articles/3"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "width": "initial",
              }
            }
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            >
              <img
                alt=""
                aria-hidden={true}
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjQwIiBoZWlnaHQ9IjM2MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              />
            </span>
            <img
              alt="Cover Image for 栄養たっぷり小松菜と油揚げのおひたしの作り方"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": "cover",
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
            <noscript>
              <img
                alt="Cover Image for 栄養たっぷり小松菜と油揚げのおひたしの作り方"
                data-nimg="intrinsic"
                decoding="async"
                loading="lazy"
                src="/_next/image?url=%2Fimages%2F4966941_s.jpg&w=1920&q=75"
                srcSet="/_next/image?url=%2Fimages%2F4966941_s.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2F4966941_s.jpg&w=1920&q=75 2x"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": "cover",
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
            </noscript>
          </span>
        </a>
      </div>
      <h2
        className="mb-4 text-2xl font-bold"
      >
        <a
          href="/articles/3"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          栄養たっぷり小松菜と油揚げのおひたしの作り方
        </a>
      </h2>
      <p
        className="dark:text-gray-300 line-clamp-2"
      >
        小松菜と油揚げのおひたしは、茹でた子大豆もやしと焼いた油揚げを混ぜた、栄養たっぷりの料理です。かつおぶしを乗せ、オリーブオイル、レモン果汁、醤油をかけてお召し上がりください。前菜やビールのおつまみにピッタリです♪
      </p>
    </article>
  </li>
  <li>
    <article>
      <div
        className="mb-4"
      >
        <a
          aria-label="熊本県産玉ねぎを使った絶品タコライスの作り方"
          href="/articles/4"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "width": "initial",
              }
            }
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            >
              <img
                alt=""
                aria-hidden={true}
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjQwIiBoZWlnaHQ9IjM2MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              />
            </span>
            <img
              alt="Cover Image for 熊本県産玉ねぎを使った絶品タコライスの作り方"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": "cover",
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
            <noscript>
              <img
                alt="Cover Image for 熊本県産玉ねぎを使った絶品タコライスの作り方"
                data-nimg="intrinsic"
                decoding="async"
                loading="lazy"
                src="/_next/image?url=%2Fimages%2F3129120_s.jpg&w=1920&q=75"
                srcSet="/_next/image?url=%2Fimages%2F3129120_s.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2F3129120_s.jpg&w=1920&q=75 2x"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": "cover",
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
            </noscript>
          </span>
        </a>
      </div>
      <h2
        className="mb-4 text-2xl font-bold"
      >
        <a
          href="/articles/4"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          熊本県産玉ねぎを使った絶品タコライスの作り方
        </a>
      </h2>
      <p
        className="dark:text-gray-300 line-clamp-2"
      >
        タコライスは沖縄県の料理ですが、今では日本のどこのご家庭でも食されるようになりました。この「絶品タコライス」は、熊本県産の完全無農薬サラダ玉ねぎを使用しており、この玉ねぎの甘さとタコライスに入ったチリパウダーの辛さが口の中で絶妙にマッチします。ご飯ものだけどビールとの相性は抜群！子供たちも大好きな料理です♪
      </p>
    </article>
  </li>
</ul>
`;
